src/commands/import-resolver.ts

```diff
@@ -77,12 +77,16 @@ export function createImportResolver({
       return spec;
     }
     if (dependencyImportMap && dependencyImportMap.imports[spec]) {
+      // if baseURL is remote, handle that outside of path.posix.join()
+      const PROTOCOL_REGEX = /^https?:\/\//;
+      const protocolMatch = config.buildOptions.baseUrl.match(PROTOCOL_REGEX);
+      const protocol = (protocolMatch && protocolMatch[0]) || '';
+      const baseUrl = config.buildOptions.baseUrl.replace(PROTOCOL_REGEX, '');
+
       let resolvedImport = isDev
         ? path.posix.resolve(webModulesPath, dependencyImportMap.imports[spec])
-        : path.posix.join(
-            config.buildOptions.baseUrl,
-            webModulesPath,
-            dependencyImportMap.imports[spec],
+        : protocol.concat(
+            path.posix.join(baseUrl, webModulesPath, dependencyImportMap.imports[spec]),
```

Comment:
Hooray for centralized resolution logic! 🎉 

Though this is a bit messy; happy for feedback on implementation details here.

Reply:
I _think_ this should work for both build and dev. The test is for build, but below, the `isDev` bypasses this logic so this doesn’t change that current behavior.

I have some feeling there may be a few weird things around having a `baseUrl` with a protocol when it comes to build & dev, but I think that’s more of a niche case we can address later.

Reply:
Definitely a little confused here, why `protocol.concat`? doesn't that return an array when we need a string?
If that's intentional, a comment would help.

Reply:
It returns a string, but you’re right it’s a bit weird. I guess I can go with a string template here instead.

I have a weird thing with ` + ` I just always feel like it’s going to betray me